features/admin.applications.v1/components/edit-application.tsx

```diff
@@ -1080,160 +529,760 @@ export const EditApplication: FunctionComponent<EditApplicationPropsInterface> =
+            <Info
+                appId={ application.id }
+                inboundProtocols={ application?.inboundProtocols }
+                isOIDCConfigLoading={ isOIDCConfigsLoading }
+                isSAMLConfigLoading={ isSAMLConfigsLoading }
+                templateId={ application?.templateId }
+                data-componentid={ `${ componentId }-server-endpoints` }
+            />
+        </ResourceTab.Pane>
+    );
+
+    /**
+     * Renders a dynamic application edit tab pane.
+     *
+     * @param tab - The metadata for the tab.
+     * @returns The rendered tab pane.
+     */
+    const DynamicApplicationEditTabPane = (tab: ApplicationEditTabMetadataInterface): ReactElement => {
+        const firstProtocolName: string = mapProtocolTypeToName(application?.inboundProtocols?.[0]?.type);
+
+        return (
+            <ResourceTab.Pane controlledSegmentation>
+                <ApplicationEditForm
+                    tab={ tab }
+                    application={ application }
+                    protocolName={ firstProtocolName }
+                    inboundProtocolConfigurations={ inboundProtocolConfig?.[firstProtocolName] }
+                    isLoading={ isLoading }
+                    onUpdate={ handleApplicationUpdate }
+                    onProtocolUpdate = { handleProtocolUpdate }
+                    readOnly={ readOnly }
+                />
+            </ResourceTab.Pane>
+        );
+    };
+
+    /**
+     * Renders a markdown guide tab pane.
+     *
+     * @param guideContent - Content to display in Markdown format.
+     * @returns The rendered tab pane.
+     */
+    const MarkdownGuideTabPane = (guideContent: string): ReactElement => {
+        const firstProtocolName: string = mapProtocolTypeToName(application?.inboundProtocols?.[0]?.type);
+
+        return (
+            <ResourceTab.Pane controlledSegmentation>
+                <ApplicationMarkdownGuide
+                    application={ application }
+                    inboundProtocolConfigurations={ inboundProtocolConfig?.[firstProtocolName] }
+                    content={ guideContent }
+                    isLoading={ isLoading }
+                    protocolName={ firstProtocolName }
+                />
+            </ResourceTab.Pane>
+        );
+    };
+
+    /**
+     * Resolves the tab panes based on the application config.
+     *
+     * @returns Resolved tab panes.
+     */
+    const resolveTabPanes = (): ResourceTabPaneInterface[] => {
+        const panes: ResourceTabPaneInterface[] = [];
+        const extensionPanes: ResourceTabPaneInterface[] = [];
+
+        if (!tabPaneExtensions && applicationConfig.editApplication.extendTabs
+            && application?.templateId !== ApplicationManagementConstants.CUSTOM_APPLICATION_OIDC
+            && application?.templateId !== ApplicationManagementConstants.CUSTOM_APPLICATION_PASSIVE_STS
+            && application?.templateId !== ApplicationManagementConstants.CUSTOM_APPLICATION_SAML) {
```

Comment:
We can define a reusable array for the custom templates and update the condition as follows for better readability.

```js
const primaryApplicationTemplates = [OIDC, PASSIVE_STS, SAML]

if (!primaryApplicationTemplates.includes(application?.templateId)) {
	/** Your logic */
}
```

Reply:
This is an existing code block. https://github.com/wso2/identity-apps/blob/05605b9367987c95ce06add63ce0e95494a90b2d/features/admin.applications.v1/components/edit-application.tsx#L234-L236